### ISSUE TITLE
SARIF log format improvements

### ISSUE THREAD

**Issue by yongyan-gh** [psastras/sarif-rs#158]
Hello,

We are the team working on SARIF eco-systems and looking for the SARIF log support for rust-clippy analysis tool. 

I created Github workflow calls rust clippy tool to generate analysis results and convert to SARIF log using clippy-sarif. It works like a charm!

Checked the generated SARIF log it's great! Have couple suggestions in terms of SAIRF spec, hope we can make the rust-clippy SARIF log better.

1. The SARIF log file does not contain a '$schema' property. This enables IDEs to provide Intellisense for SARIF log files. Consider adding a '$schema' property that refers to the final, OASIS standard version of the SARIF 2.1.0 schema. E.g.
```
{
  "version": "2.1.0",
  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
  "runs": [ ... ]
}
```
2. The tool 'clippy' does not provide 'informationUri'. This property helps the developer responsible for addessing a result by providing a way to learn more about the tool. It can be url of rust-clippy repo or web site. E.g.
```
{
  ...
    "tool": {
      "driver": {
        "name": "clippy",
        "informationUri": "https://rust-lang.github.io/rust-clippy/"
      }
    }
  ...
}
```
3. Rules in log do not provide a help URI. Providing a URI where users can find detailed information about the rule helps users to understand the result and how they can best address it. The full description contains a link to the page for detail rule information
    > for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons

    But better to set "helpUri" property, it helps SARIF log viewers like Github security tab to generate hyper link for each result. E.g.
```
          "rules": [
            {
              "id": "clippy::absurd_extreme_comparisons",
              "helpUri": "https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons"
              "fullDescription": { ... }
            },
```

4. Rule "id" property does not follow the recommended format: a short string identifying the tool concatenated with a numeric rule number, for example, 'CS2001'. Using a conventional format for the rule id provides a more uniform experience across tools. This may depend on how Rust-clippy defines rules. If has to use same value for rule "id" and "name", the "name" property can be removed.

5. Its recommended to place "version" property at beginning of SARIF log, although the order in which properties appear in a JSON object value is not semantically significant. This will make it easier for parsers to handle multiple versions of the SARIF format if new versions are defined in the future.

Please take a look the points above and let's know if any question. 

Thanks!

cc @eddynaka @michaelcfanning @josepalafox

**Comment by josepalafox** [comment]
Could you add the workflow for rustclippy here - https://github.com/actions/starter-workflows

It'll make it more discoverable in the UI

**Comment by psastras** [comment]
@yongyan-gh ,

Hi, thanks for the recommendations and appreciate the feedback.

1, 2, 3 should all be available in the latest version (0.3.0).

For 4. I have removed the "name" property as suggested and it is also updated in version 0.3.0. Clippy does not provide rule IDs conforming with the SARIF standard, so the abbreviated rule name is used for the ID value. If you are interested in a more standard ID value, that would probably need to be done in clippy or rustc (and would be a bit involved I imagine).

For 5. I've broken it out int #161 since it's a bit more involved and not semantically significant.

Since the remaining work here should be captured now in #161 I will close this issue. Feel free to re-open the issue if there are other concerns.
